random_fields/geostatistical_cpt_interpretation.py

```diff
@@ -0,0 +1,682 @@
+                 length_scale_bounds: List[float] = [1.0, 1000.],
+                 lenght_scale_bounds_vertical: List[float] = [0.01, 10.],
+                 custom_kernel=None) -> None:
+        """
+        Initiates instance and sets kernel to WhiteKernel() + RBF() if kernel is not provided
+
+        Args:
+            - nb_dimensions (int): number of physical dimensions.
+            - v_dim (int): Dimension index of the vertical scale of fluctuation
+            - lenght_scale (float): length scale of the kernel. Defaults to 100
+            - length_scale_bounds (List[float]): bounds of the length scale. Defaults to [1.0, 1000.]
+            - lenght_scale_bounds_vertical (List[float]): bounds of the vertical length scale. Defaults to [0.01, 10.]
+            - custom_kernel (:class:`sklearn.gaussian_process.kernels.Kernel`): Custom spatial correlation kernel.
+                Defaults to None.
+
+        Raises:
+            TypeError: if `kernel` is not a class from module `sklearn.gaussian_process.kernels`
+        """
+        self.nb_dimensions = nb_dimensions
+        self.v_dim = v_dim
+
+        self.data_coords = np.empty(0)
+        self.data_values = np.empty(0)
+        self.noise_level = np.empty(0)
+        self.vertical_scale_fluctuation = np.empty(0)
+        self.anisotropy = np.empty(0)
+
+        # check if `kernel` has the correct parent class
+        if custom_kernel:
+            if custom_kernel.__module__ != 'sklearn.gaussian_process.kernels':
+                raise TypeError("`kernel` should an instance of a class from module `sklearn.gaussian_process.kernels`")
+
+        # if kernel does not exist, create a default
+        if not custom_kernel:
+            length_scale_kernel = length_scale * np.ones(nb_dimensions)
+            length_scale_bounds_kernel = [length_scale_bounds] * nb_dimensions
+            length_scale_kernel[v_dim] = self.v_dim
+            length_scale_bounds_kernel[v_dim] = lenght_scale_bounds_vertical
+
+            kernel = WhiteKernel(0.01) + RBF(length_scale=length_scale_kernel,
+                                             length_scale_bounds=length_scale_bounds_kernel)
+        self.gpr = GPR(kernel=kernel)
+
+    def calibrate(self, coords: npt.NDArray[np.float64], values: npt.NDArray[np.float64]) -> None:
+        """
+        Calibrates the geostatistical model
+
+        Args:
+            - coords (npt.NDArray[np.float64]): data spatial coordinates
+            - values (npt.NDArray[np.float64]): data values
+
+        Raises:
+            - Warning: if `coords` is transposed to match dimensions (n_points,n_dims)
+            - Exception: if shape of argument 'coords' incompattible with dimensionality
+
+        """
+
+        if coords.shape[0] == self.nb_dimensions and coords.shape[1] != self.nb_dimensions:
+            self.data_coords = coords.T
+            raise Warning('`coords` is transposed to match dimensions (n_points,n_dims)')
```

Comment:
warnings are not raised like this, but:
import warnings

warnings.warn("sdadsa")